core-rust/core-api-server/src/core_api/handlers/stream_transactions.rs

```diff
@@ -47,22 +47,30 @@ pub(crate) async fn handle_stream_transactions(
 
     let max_state_version = database.max_state_version();
 
-    let txns = database.get_committed_transaction_bundles(
+    let transactions = database.get_committed_transaction_bundles(
         from_state_version,
         limit.try_into().expect("limit out of usize bounds"),
     );
 
-    let api_txns = txns
-        .into_iter()
-        .map(|(ledger_transaction, receipt, identifiers)| {
-            Ok(to_api_committed_transaction(
-                &mapping_context,
-                ledger_transaction,
-                receipt,
-                identifiers,
-            )?)
-        })
-        .collect::<Result<Vec<models::CommittedTransaction>, ResponseError<()>>>()?;
+    // Reserve enough for the "header" fields
+    let mut current_total_size = 2048;
+    let mut api_txns = Vec::new();
+    for (ledger_transaction, receipt, identifiers) in transactions {
+        let committed_transaction = to_api_committed_transaction(
+            &mapping_context,
+            ledger_transaction,
+            receipt,
+            identifiers,
+        )?;
+
+        let committed_transaction_size = committed_transaction.get_json_size();
+        if current_total_size + committed_transaction_size > MAX_STREAM_TOTAL_SIZE_PER_RESPONSE {
+            break;
```

Comment:
Guess we should at least log this information (produce some metric)? This feature seems extremely fragile so I'd invest a bit in observability here.

Reply:
We don't have any real metrics in core-api right now.

Added `warn!`s.

Reply:
Might be nice to have metrics - we could add CoreApiMetrics (registered with the state manager registry on Core API boot-up) to the `CoreApiState`.

But let's save this for a follow-up PR 👍